sirepo/job_supervisor.py

```diff
@@ -758,6 +758,19 @@ async def _receive_api_statefulCompute(self, req):
     async def _receive_api_statelessCompute(self, req):
         return await self._send_simulation_compute(req)
 
+    async def _receive_api_wakeAgent(self, req):
+        c = self._create_op(job.OP_WAKEUP, req)
+        try:
+            await c.prepare_send()
+        except Awaited:
+            # OPTIMIZATION: _agent_ready is the first thing that could raise Awaited.
+            # In the event that it does, the agent is still started,
```

Comment:
I'm on the fence. We originally had a `POSIT` but changed to `OPTIMIZATION`. We could do the normal awaited loop around prepare_send which would eliminate the need for a comment. But, we are just expecting one awaited since the agent starting is the first part of prepare send. This saves us from using the loop and waiting for it to terminate which is an optimization.

`Agent is running so nothing more to do.` Why say this? I don't think we know if the agent is running. For example, let's say we can't start the agent (bug in agent code). We would still get to this code.

Reply:
If the agent fails because of the agent code, that's a different dimension, just like if we receive the message at the same time job_supervisor is started. The entire process is an optimization, because it works without doing anything.

I think there's a larger problem here: two agents need to be started. I think the job_supervisor has to start a task asynchronously on behalf of the user, not a job.